test/gtest/ucm/malloc_hook.cc

```diff
@@ -353,6 +353,39 @@ void test_thread::mem_event(ucm_event_type_t event_type, ucm_event_t *event)
     pthread_mutex_unlock(&m_stats_lock);
 }
 
+static void *perform_mmap(size_t size, char fill_value) {
+    void *ptr = mmap(NULL, size,
+                     PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS,
+                     -1, 0);
+    EXPECT_NE(MAP_FAILED, ptr);
+    memset(ptr, fill_value, size);
+    return ptr;
+}
+
+static void *perform_mremap(void *old_ptr, size_t old_size, size_t new_size,
+                            void *new_addr) {
+    void *new_ptr = mremap(old_ptr, old_size, new_size,
+                           MREMAP_MAYMOVE | MREMAP_FIXED, new_addr);
+    EXPECT_NE(MAP_FAILED, new_ptr) << strerror(errno);
+    return new_ptr;
+}
+
+static void set_and_check_env_variable(pthread_mutex_t* lock,
```

Comment:
can it be static function inside test_thread class?

Reply:
I'm not sure we have to make this change, checking

Reply:
No need for this commit since https://github.com/openucx/ucx/pull/10131/commits/420caff458f0e3b11fcdb2052c179980fc9d5fdf